### ISSUE TITLE
System.IO.Ports.Tests.OpenDevices.OpenDevices01 failed in uap

### ISSUE THREAD

**Issue by safern** [dotnet/corefx#21156]
This test is throwing the following exception: 

```
System.Exception : Unknown Win32 Error: 5
        Stack Trace:
             at System.IO.Ports.Tests.DosDevices.CallQueryDosDevice(String name, Int32& dataSize)
             at System.IO.Ports.Tests.DosDevices.Initialize()
             at System.IO.Ports.Tests.DosDevices..ctor()
             at System.IO.Ports.Tests.OpenDevices.OpenDevices01()
```

When calling [CallQueryDosDevice](https://github.com/dotnet/corefx/blob/master/src/System.IO.Ports/tests/SerialPort/DosDevices.cs#L133)

cc: @danmosemsft @JeremyKuhne

**Comment by JeremyKuhne** [comment]
We've got a new API coming from Windows for this.

**Comment by safern** [comment]
Great, because the tests are hanging also because of this when calling: https://github.com/dotnet/corefx/blob/master/src/System.IO.Ports/tests/Support/PortHelper.cs#L20-L76 through conditional fact. 

I'm adding `if (PlatformDetection.IsWinRT) return new string[0];` at the beginning of that helper method to unblock test execution.

**Comment by danmosemsft** [comment]
@safern For now it looks like we won't get an enumeration API that works in UWP. It seems like we have to

1. Verify that aside from GetPortNames(), assuming CreateFile is replaced with CreateFileFromApp, serial port functionality works in a UWP (ie., works within AppContainer)
1. If it does, hard code GetPortNames() when in AppContainer, to return COM1 through COMn for some n. (For those ports that don't actually exist there would be a failure downstream.)
1. Update our package.

**Comment by danmosemsft** [comment]
@JeremyKuhne @willdean for above note. We'd need someone with actual serial ports to try this -- I think this is lower than other stuff on Jeremy's priority list right now.

**Comment by willdean** [comment]
@danmosemsft I have allowed all things UWP to pass me by completely, so I'm not really clear what's needed to test this - can it be tested on a Win10 machine by invoking the test suite differently?  

What would be the minimum OS requirement (I have machines with various combinations of ports, but they have various versions of Windows too)

**Comment by danmosemsft** [comment]
@safern at this point, there's no setup required anymore right? Just building from the root with `build -framework:uap` then running the test with `/p:targetgroup=uap /t:buildandtest` should work.

**Comment by danmosemsft** [comment]
Changing the milestone as serial ports is not essential to ship UWP6.0, it's a "would like"

**Comment by safern** [comment]
> @safern at this point, there's no setup required anymore right? Just building from the root with build -framework:uap then running the test with /p:targetgroup=uap /t:buildandtest should work.

Correct. That's enough

**Comment by danmosemsft** [comment]
@willdean as for the OS, I think we require RS2, although it may work with RS1 if you don't run tests that do "remote execute". RS2 is the Creators Edition, it's a free upgrade.

**Comment by pjanotti** [comment]
There is also the path via another API to be added to UWP but that API is not ready yet.

**Comment by danmosemsft** [comment]
@pjanotti are we still waiting on Windows? What remains to get serial ports working for UWP?

**Comment by pjanotti** [comment]
@danmosemsft I have yet to get a Windows build with the working version of the API - I will see where it is now.

**Comment by danmosemsft** [comment]
OK. If we decide we can't do this in the near term I'd like to do so consciously (although as I mentioned it seems our library doesn't need to be on the exact same schedule, if it's blocked on Windows, we need to get that fix into RS3)

**Comment by pjanotti** [comment]
It should be part of RS3 but for now it is not part of prerelease yet.

**Comment by danmosemsft** [comment]
@pjanotti I forgot to ask -- presumably using this API will require us to add a uap configuration to this library? 
Either way, if it needs a code change to work on UWP, you should make sure @joperezr knows, as we will need to ship it. As it is, it probably won't ship from the UWP branch since it's not UWP specific currently.

**Comment by pjanotti** [comment]
@danmosemsft and @joperezr  Yes, this requires UAP configuration since we need to use a different API.

**Comment by joperezr** [comment]
that is a good catch @danmosemsft, as he points out, the way we have it now means that we won't be shipping this binary for UWP since it's not part of the metapackage. In case we do want to ship System.IO.Ports for UAP then I would suggest adding the uap configuration now and then consider to add this to the metapackage since we probably won't want to publish this as it's own package (cc: @weshaggard )

**Comment by pjanotti** [comment]
@joperezr I still do not have a stable build to validate this but I can start a PR soon (there is one last issue that I need to fix). That way we can be ready as soon as I am able to run the tests.

**Comment by pjanotti** [comment]
@JeremyKuhne @willdean I have minimally tested code (i.e.: open serial port via new API) to handle the UWP case but I get very low coverage since the tests are deadlocking when I have loopback or nullmodem in place. The deadlock is not related to the change or anything in corefx since it happens with a minimal repro with the full framework. I tried the MS driver and the manufacturer driver to same result. I will open a separate issue to investigate that, meanwhile I will start a PR so you guys can give some feedback.

**Comment by pjanotti** [comment]
I opened a separate issue #22966 to track the problems that I am having to better test serial ports (it affects not only the UWP change).

**Comment by danmosemsft** [comment]
Still shooting for 6.0 so I want to track it.

**Comment by pjanotti** [comment]
PR to get it on UWP6.0 #23261

**Comment by pjanotti** [comment]
@danmosemsft time to close this one...
